R/utility.R

```diff
@@ -127,6 +80,64 @@ waitForNodesToComplete <- function(poolId, timeout = 86400) {
   stop("Timeout expired")
 }
 
+.processNodeCount <- function(nodes) {
+  nodesWithFailures <- c()
+  currentNodeCount <- 0
+  for (i in 1:length(nodes$value)) {
+    # The progress total count is the number of the nodes. Each node counts as 1.
+    # If a node is not in idle, prempted, running, or start task failed, the value is
+    # less than 1. The default value is 0 because the node has not been allocated to
+    # the pool yet.
+    nodeValue <- switch(
+      nodes$value[[i]]$state,
+      "idle" = {
+        1
+      },
+      "creating" = {
+        0.25
+      },
+      "starting" = {
+        0.50
+      },
+      "waitingforstartask" = {
+        0.75
+      },
+      "starttaskfailed" = {
+        nodesWithFailures <- c(nodesWithFailures, nodes$value[[i]]$id)
+        1
+      },
+      "preempted" = {
+        1
+      },
+      "running" = {
+        1
+      },
+      0
+    )
+
+    currentNodeCount <-
+      currentNodeCount + nodeValue
+  }
+  return(list(currentNodeCount = currentNodeCount, nodesWithFailures = nodesWithFailures))
+}
+
+.showNodesFailure <- function(nodesWithFailures) {
```

Comment:
We should print a brief message of the start task failure so users don't have to find the logs